hazelcast/src/test/java/com/hazelcast/internal/dynamicconfig/DynamicConfigTest.java

```diff
@@ -670,6 +672,29 @@ private void assertConfigurationsEqualOnAllMembers(PNCounterConfig config) {
         }
     }
 
+    @Test
+    public void testExternalDataStoreConfig() {
+        Properties properties = new Properties();
+        properties.setProperty("prop1", "val1");
+        properties.setProperty("prop2", "val2");
+        ExternalDataStoreConfig externalDataStoreConfig = new ExternalDataStoreConfig()
+                .setName("some-name")
+                .setClassName("some-class-name")
+                .setProperties(properties);
+
+
+        driver.getConfig().addExternalDataStoreConfig(externalDataStoreConfig);
+        assertConfigurationsEqualOnAllMembers(externalDataStoreConfig);
+    }
+
+    private void assertConfigurationsEqualOnAllMembers(ExternalDataStoreConfig expectedConfig) {
```

Comment:
nit, not related, but all methods below can be simplified to:

```java
private <T> void assertConfigurationsEqualOnAllMembers(T expectedConfig, Function<T, Config, T> copyGetter) {
for (HazelcastInstance instance : members) {
ExternalDataStoreConfig registeredConfig = copyGetter.apply(expectedConfig, instance.getConfig());
assertThat(registeredConfig).isEqualTo(expectedConfig);
}
}
```

then all of them become one-liners like that:
```java
private void assertConfigurationsEqualOnAllMember(CacheSimpleConfig config){
return assertConfigurationsEqualOnAllMembers(config, (tc, c) -> c.getCacheConfig(config.getName()));
        }
```

Reply:
Good idea! fixed: https://github.com/hazelcast/hazelcast/pull/21930/commits/0009c154ea4ece21f4569b36c036ade2e5feb7a8